llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -6751,6 +6806,24 @@ Value *SPIRVToLLVM::transSPIRVImageWriteFromInst(SPIRVInstruction *bi, BasicBloc
   addr[lgc::Builder::ImageAddressIdxCoordinate] = transValue(bii->getOpValue(opndIdx++), bb->getParent(), bb);
   Value *texel = transValue(bii->getOpValue(opndIdx++), bb->getParent(), bb);
 
+  // Create smaller texel vector, if num texel components > num channels. This
+  // ensures that no unused channels are being stored.
+  if (auto texelVecType = dyn_cast<FixedVectorType>(texel->getType())) {
+    unsigned numChannels = getImageNumChannels(imageInfo.desc);
+    if (numChannels > 0 && numChannels < texelVecType->getNumElements()) {
+      if (numChannels == 1) {
+        texel = getBuilder()->CreateExtractElement(texel, getBuilder()->getInt32(0));
+      } else {
+        SmallVector<int, 3> channels;
+        for (int channel = 0; channel < numChannels; ++channel) {
+          channels.push_back(channel);
+        }
+
+        texel = getBuilder()->CreateShuffleVector(texel, llvm::makeArrayRef(channels));
```

Comment:
Don't need `llvm::`. Don't need `makeArrayRef` in modern C++, you can just use the `ArrayRef` constructor. (Do you even need it at all or can you just pass in `channels`?)

Reply:
Passing in `channels` directly also works. I think the call to makeArrayRef is some relict from a previous revision.